reader/src/main/scala/za/co/absa/atum/reader/FlowReader.scala

```diff
@@ -16,9 +16,24 @@
 
 package za.co.absa.atum.reader
 
-class FlowReader {
-  def foo(): String = {
-    // just to have some testable content
-    "bar"
-  }
+import sttp.client3.SttpBackend
+import sttp.monad.MonadError
+import za.co.absa.atum.model.types.basic.AtumPartitions
+import za.co.absa.atum.reader.basic.ReaderWithPartitioningId
+import za.co.absa.atum.reader.server.ServerConfig
+
+/**
+ * This class is a reader that reads data tight to a flow.
+ * @param mainFlowPartitioning  - the partitioning of the main flow; renamed from ancestor's 'flowPartitioning'
+ * @param serverConfig          - tha Atum server configuration
+ * @param backend               - sttp backend, that will be executing the requests
+ * @param ev                    - using evidence based approach to ensure that the type F is a MonadError instead of using context
+ *                              bounds, as it make the imports easier to follow
+ * @tparam F                    - the effect type (e.g. Future, IO, Task, etc.)
+ */
+class FlowReader[F[_]](val mainFlowPartitioning: AtumPartitions)
```

Comment:
Sure, that I understand, but I wonder why it needs to be public val? It would be reasonable if you were going to use the trait `PartitioningIdProvider` and by using public `val partitioning` in the constructor you provided the requirement for `def partitioning: AtumPartitions`.